Source/Dafny/Resolver.cs

```diff
@@ -10042,6 +10068,41 @@ public class ResolveTypeReturn
       ConstrainTypeExprBool(s.Expr, "type of RHS of assign-such-that statement must be boolean (got {0})");
     }
 
+    // TODO is there already a library of convenience AST construction methods like this one?
```

Comment:
There are various routines like this sprinkled across the source code. Unfortunately, they have been written in an ad-hoc fashion when the need arose. I don't think this particular one exists elsewhere.